abis/IAllowList.abi

```diff
@@ -0,0 +1 @@
+[{"inputs":[{"internalType":"address","name":"addr","type":"address"}],"name":"readAllowList","outputs":[{"internalType":"uint256","name":"role","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"addr","type":"address"}],"name":"setAdmin","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"addr","type":"address"}],"name":"setEnabled","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"addr","type":"address"}],"name":"setNone","outputs":[],"stateMutability":"nonpayable","type":"function"}]
```

Comment:
Do we need these ABI files for `IAllowList.sol` and `IRewardManager.sol` ?

We should also consider adding a test to CI to ensure that the ABI we have in the code matches what we get by compiling the correct smart contract with the correct version of the Solidity compiler since this is difficult to review manually. Not in scope for this PR.

Reply:
nope certainly do not need